src/app/saga.ts

```diff
@@ -93,14 +94,19 @@ const DO_NOT_LOCK_PERIOD = 30000 // 30 sec
 // Work that's done before other sagas are initalized
 // Be mindful to not put long blocking tasks here
 export function* appInit() {
-  yield call(initializeSentry)
+  const t0 = Date.now()
+  yield all([call(initializeSentry), call(ValoraAnalytics.init)])
+
   // This step is important if the user if offline and unable to fetch remote
   // config values, we can use the persisted value instead of an empty one
   const sentryNetworkErrors = yield select(sentryNetworkErrorsSelector)
   Logger.setNetworkErrors(sentryNetworkErrors)
 
   const supportedBiometryType = yield call(Keychain.getSupportedBiometryType)
   yield put(setSupportedBiometryType(supportedBiometryType))
+
+  const t1 = Date.now()
+  ValoraAnalytics.track(AppEvents.app_tooling_initialized, { durationMs: t1 - t0 })
```

Comment:
this seems like a bit of a hack to get performance profiling in this one place. I'd prefer if product analytics and performance analytics were more separate and used domain-specific tooling for each. Could we try to use Sentry profiling for this instead?

Reply:
it is kind of 😅 i guess i want to understand how much of a performance hit our users are incurring. but i don't trust in our understanding of sentry - it's easy enough to put sentry instrumentation in here but we've seen all kinds of weird reporting in sentry. idk, i can also use sentry for this but i'd like to keep the analytics event either way. it seems useful for measuring MAU's (since we don't want to use the "fetch balances" one)

Reply:
We actually already have a a custom instrumentation implemented in #2728 that we could use.

Reply:
@satish-ravi this will only be triggered for a cold start, if the app has been backgrounded for long enough and been terminated by the device then it will be run. but for a quick background - foreground switch then it will not run.

it seems kind of wild to use sentry to measure sentry's init time, it didn't log any errors so i put it in since everyone feels so strongly about it. but i would like to keep the analytics event in as a sanity check, in case the sentry data does not make sense

Reply:
fwiw i do believe in using sentry, but recently i've had a hard time understanding and trusting the data that we see in the dashboards. so i was reluctant to throw more data there that we might not understand, but maybe it's just me who can't make sense of the data. it'd be nice to have a deep dive brownbag if anyone has more knowledge (for example, the dapps screen has a p95 value of [over 1 minute](https://valora-inc.sentry.io/performance/summary/?project=1250733&query=&referrer=performance-transaction-summary&statsPeriod=14d&transaction=DAppsExplorerScreen&unselectedSeries=p100%28%29) but we cache the results so there should be virtually no load time for the user...)